src/generators/javascript/JavaScriptRenderer.ts

```diff
@@ -3,7 +3,201 @@ import { JavaScriptGenerator, JavaScriptOptions } from './JavaScriptGenerator';
-
+//Based on https://www.w3schools.com/js/js_reserved.asp
+const reservedJavaScriptKeywords = [
+  //Standard reserved keywords
+  'abstract',
+  'arguments',
+  'await',
+  'boolean',
+  'break',
+  'byte',
+  'case',
+  'catch',
+  'char',
+  'class',
+  'const',
+  'continue',
+  'debugger',
+  'default',
+  'delete',
+  'do',
+  'double',
+  'else',
+  'enum',
+  'eval',
+  'export',
+  'extends',
+  'false',
+  'final',
+  'finally',
+  'float',
+  'for',
+  'function',
+  'goto',
+  'if',
+  'implements',
+  'import',
+  'in',
+  'instanceof',
+  'int',
+  'interface',
+  'let',
+  'long',
+  'native',
+  'new',
+  'null',
+  'package',
+  'private',
+  'protected',
+  'public',
+  'return',
+  'short',
+  'static',
+  'super',
+  'switch',
+  'synchronized',
+  'this',
+  'throw',
+  'throws',
+  'transient',
+  'true',
+  'try',
+  'typeof',
+  'var',
+  'void',
+  'volatile',
+  'while',
+  'with',
+  'yield',
+  // Reserved for > ECMAScript 5/6
+  'abstract', 
+  'boolean', 
+  'byte', 
+  'char',
+  'double', 
+  'final', 
+  'float', 
+  'goto',
+  'int', 
+  'long', 
+  'native', 
+  'short',
+  'synchronized', 
+  'throws', 
+  'transient', 
+  'volatile',
+  //Reserved built-in objects, properties and methods
+  'hasOwnProperty', 
+  'Infinity', 
+  'isFinite', 
+  'isNaN',
+  'isPrototypeOf', 
+  'length', 
+  'Math', 
+  'NaN',
+  'name', 
+  'Number', 
+  'Object', 
+  'prototype',
+  'String', 
+  'toString', 
+  'undefined', 
+  'valueOf',
+  //Java reserved words
+  'getClass', 
+  'java', 
+  'JavaArray', 
+  'javaClass',
+  'JavaObject', 
+  'JavaPackage',
+  //Other reserved words
+  'alert', 
+  'all', 
+  'anchor', 
+  'anchors',
+  'area', 
+  'assign', 
+  'blur', 
+  'button',
+  'checkbox', 
+  'clearInterval', 
+  'clearTimeout', 
+  'clientInformation',
+  'close', 
+  'closed', 
+  'confirm', 
+  'constructor',
+  'crypto', 
+  'decodeURI', 
+  'decodeURIComponent', 
+  'defaultStatus',
+  'document', 
+  'element', 
+  'elements', 
+  'embed',
+  'embeds', 
+  'encodeURI', 
+  'encodeURIComponent', 
+  'escape',
+  'event', 
+  'fileUpload', 
+  'focus', 
+  'form',
+  'forms', 
+  'frame', 
+  'innerHeight', 
+  'innerWidth',
+  'layer', 
+  'layers', 
+  'link', 
+  'location',
+  'mimeTypes', 
+  'navigate', 
+  'navigator', 
+  'frames',
+  'frameRate', 
+  'hidden', 
+  'history', 
+  'image',
+  'images', 
+  'offscreenBuffering', 
+  'open', 
+  'opener',
+  'option', 
+  'outerHeight', 
+  'outerWidth', 
+  'packages',
+  'pageXOffset', 
+  'pageYOffset', 
+  'parent', 
+  'parseFloat',
+  'parseInt', 
+  'password', 
+  'pkcs11', 
+  'plugin',
+  'prompt', 
+  'propertyIsEnum', 
+  'radio', 
+  'reset',
+  'screenX', 
+  'screenY', 
+  'scroll', 
+  'secure',
+  'select', 
+  'self', 
+  'setInterval', 
+  'setTimeout',
+  'status', 
+  'submit', 
+  'taint', 
+  'text',
+  'textarea', 
+  'top', 
+  'unescape', 
+  'untaint',
+  'window'
```

Comment:
These same we should use also in TS. Also please move the all arrays (also for other langs) to the separate file like `constants.ts`:
- generators/typescript/constant.ts
- generators/java/constant.ts

etc.

Reply:
> These same we should use also in TS.
We should? Thought the transpile process would ensure compatibility there?

Reply:
Nope, for example TS `window` will be `window` in transpiled JS, so we should also treat the below keywords as reserved for TS.

Reply:
Huh, fair enough 😄 Changed